File: src/typescript/src/transport/transaction/message/poly-bus-message-not-found-error.ts
import { PolyBusError } from '../../../poly-bus-error';

/**
 * Is thrown when a requested type, attribute/decorator, or header was not registered with the message system.
 */
export class PolyBusMessageNotFoundError extends PolyBusError {
  /**
   * Creates a new PolyBusMessageNotFoundError instance.
   */
  constructor() {
    super(2, 'The requested type, attribute/decorator, or header was not found.');
    this.name = 'PolyBusMessageNotFoundError';

    // Maintains proper stack trace for where our error was thrown (only available on V8)
    if (Error.captureStackTrace) {
      Error.captureStackTrace(this, PolyBusMessageNotFoundError);
    }
  }
}

File: src/python/src/transport/transaction/message/poly_bus_message_not_found_error.py
"""
Exception raised when a requested type, attribute/decorator, or header is not found.
"""
from ....poly_bus_error import PolyBusError


class PolyBusMessageNotFoundError(PolyBusError):
    """Thrown when a requested type, attribute/decorator, or header was not registered with the message system."""
    
    def __init__(self):
        super().__init__(2, "The requested type, attribute/decorator, or header was not found.")
